src/main/java/org/testng/IDataProviderListener.java

```diff
@@ -0,0 +1,25 @@
+package org.testng;
+
+/**
+ * A listener that gets invoked before and after a data provider is invoked by TestNG.
+ */
+public interface IDataProviderListener extends ITestNGListener {
+
+    /**
+     * This method gets invoked just before a data provider is invoked.
+     *
+     * @param method                  - The {@link ITestNGMethod} method that is going to consume the data
+     *                                provided by the data provider.
+     * @param dataProviderInformation - A {@link DataProviderInformation} object that contains details about the
+     *                                data provider that is about to be executed.
+     */
+    void beforeDataProviderExecution(ITestNGMethod method, DataProviderInformation dataProviderInformation);
```

Comment:
As the data provider is the main actor of the class, maybe it should be the first param of the methods :)